Ferries shouldn't be set to destination only.

knaveofdiamonds opened the pull request:
It doesn't make sense for ferries to be destination only, and it causes issues when people tag them as `access=customers` (which, despite the wiki suggesting avoiding tagging, feels like a reasonable thing to do. I think we should be "liberal in what we accept" here). For example this car route: https://valhalla.openstreetmap.de/directions?profile=car&wps=-1.948528289794922%2C50.68357073782352%2C-1.95157527923584%2C50.67737071648344

## Tasklist

 - [x] Add tests
 - [ ] Add #fixes with the issue number that this PR addresses
 - [ ] Update the docs with any new request parameters or changes to behavior described
 - [ ] Update the [changelog](CHANGELOG.md)
 - [ ] If you made changes to the lua files, update the [taginfo](taginfo.json) too
 - [ ] If you made changes to a translation file, [update transifex](docs/docs/locales.md) too

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

nilsnolde reviewed (dismissed):
Thx, makes sense.

Merging will have to wait a bit (for all PRs, not only this one) until https://github.com/valhalla/valhalla/pull/5427 is done and we can run CI again.

knaveofdiamonds commented:
> sorry, seems you need to run `./scripts/format.sh` @knaveofdiamonds

Done for the files changed in this PR. FWIW, I get quite a lot of files changed when I run that:

	modified:   src/meili/transition_cost_model.cc
	modified:   src/midgard/logging.cc
	modified:   src/mjolnir/linkclassification.cc
	modified:   src/mjolnir/pbfgraphparser.cc
	modified:   src/sif/autocost.cc
	modified:   src/sif/bicyclecost.cc
	modified:   src/sif/dynamiccost.cc
	modified:   src/sif/motorcyclecost.cc
	modified:   src/sif/motorscootercost.cc
	modified:   src/sif/pedestriancost.cc
	modified:   src/sif/truckcost.cc
	modified:   src/skadi/sample.cc
	modified:   src/thor/costmatrix.cc
	modified:   src/thor/expansion_action.cc
	modified:   test/gurka/test_ferry_connections.cc
	modified:   test/gurka/test_ignore_restrictions.cc
	modified:   test/gurka/test_indoor.cc
	modified:   test/gurka/test_recost.cc
	modified:   test/hierarchylimits.cc
	modified:   third_party/googletest (new commits)
	modified:   valhalla/baldr/openlr.h
	modified:   valhalla/baldr/tilegetter.h
	modified:   valhalla/meili/traffic_segment_matcher.h
	modified:   valhalla/midgard/gridded_data.h
	modified:   valhalla/midgard/util.h
	modified:   valhalla/mjolnir/osmaccessrestriction.h

but I assume we don't want these changed as part of this change.
